fairlearn/metrics/_metric_frame.py

```diff
@@ -462,7 +504,9 @@ def __group(
         else:
             return result
 
-    def group_max(self, errors: str = "raise") -> Union[Any, pd.Series, pd.DataFrame]:
+    def group_max(
+        self, errors: str = "raise", quantiles: Optional[List[float]] = None
```

Comment:
This is my main thought for the new API @MiroDudik  - have an optional `quantiles=` argument, which will trigger the usage of the bootstrap samples computed in the constructor

Reply:
Let's discuss on #999 (or suggest an alternative)--otherwise we will have an exploding number of comments.